stdlib/public/runtime/Casting.cpp

```diff
@@ -635,23 +635,13 @@ static bool _dynamicCastToAnyHashable(OpaqueValue *destination,
 
   // If we do find one, the cast succeeds.
 
-  // The intrinsic wants the value at +1, so we have to copy it into
-  // a temporary.
-  ValueBuffer buffer;
-  bool mustDeallocBuffer = false;
-  if (!(flags & DynamicCastFlags::TakeOnSuccess)) {
-    auto *valueAddr = sourceType->allocateBufferIn(&buffer);
-    source = sourceType->vw_initializeWithCopy(valueAddr, source);
-    mustDeallocBuffer = true;
-  }
-
   // Initialize the destination.
   _swift_convertToAnyHashableIndirect(source, destination,
                                       sourceType, hashableConformance);
 
-  // Deallocate the buffer if we used it.
-  if (mustDeallocBuffer) {
-    sourceType->deallocateBufferIn(&buffer);
+  // Destroy the value if requested.
+  if (flags & DynamicCastFlags::TakeOnSuccess) {
+    sourceType->vw_destroy(source);
```

Comment:
The other way of doing this, by the way, would be to mark the intrinsic parameter as `__owned`.  I don't have an opinion about which one is more important to optimize.